### PR TITLE
generate executable for each rmw_implementation

### PR DESCRIPTION
**Pull request by mikaelarguedas**
fixes https://github.com/ros2/examples/issues/124
connects to https://github.com/ros2/examples/issues/124

needs https://github.com/ament/ament_tools/issues/109 to be fixed before merge

### CODE REVIEW

File: rclpy_examples/listener_py.py [hunk 1]
```diff
@@ -27,16 +27,28 @@ def chatter_callback(msg):
...
+        String, 'chatter', chatter_callback, qos_profile_default)
     assert sub  # prevent unused warning
 
     while rclpy.ok():
         rclpy.spin_once(node)
 
+
+class main_for_rmw_impl_class(object):
+    def __getattr__(self, key):
+        return main([key])
```
Comment (dirk-thomas) [inline rclpy_examples/listener_py.py]: `sys.argv` which the `args` keyword argument defaults to has a different semantic. It contains the script name as the first entry. Therefore I think this needs to be refactored somehow to have the same semantic when being invoked from the command line as well as the entry point.

**Comment by mikaelarguedas** [comment]
good points, thanks.
fixed in https://github.com/ros2/examples/pull/127/commits/f59844f44aea4819e595e1a857592776426fb50e

File: rclpy_examples/listener_py.py [hunk 2]
```diff
@@ -25,23 +25,20 @@ def chatter_callback(msg):
...
                         help='rmw_implementation identifier')
-    if args is None:
-        parsargs = parser.parse_args()
-    else:
-        parsargs = parser.parse_args(args)
+    args = parser.parse_args(argv)
 
-    if parsargs is not None:
-        if parsargs.rmw_implementation is not None:
+    if args is not None:
```
Comment (dirk-thomas) [inline rclpy_examples/listener_py.py]: Obsolete condition.

**Comment by mikaelarguedas** [comment]
moved to https://github.com/ros2/tutorials/pull/3

**Pull request closed without merging**